libindy/src/commands/anoncreds/verifier.rs

```diff
@@ -116,6 +116,15 @@ impl VerifierCommandExecutor {
                                format!("Requested predicates {:?} do not correspond to received {:?}", requested_predicates, received_predicates)));
         }
 
+        if !self.anoncreds_service.verifier.verify_requested_restrictions(&self.anoncreds_service.prover,
+                                                                                     &proof_req,
+                                                                                     &proof,
+                                                                                     schemas,
+                                                                                     cred_defs)? {
+            warn!("Requested restrictions in Proof Request don't match Proof");
+            return Ok(false);
```

Comment:
In my opinion, throwing `ProofRejected` is much better. 
 Libindy provides a function for getting error message and stack trace in addition to an error code. 
So, ProofRejected context should contain a message that describes what validation failed.
On every validation level function should return error with a message instead of bool flag. It will be much informative

Reply:
I am trying to follow the same pattern that ```self.anoncreds_service.verifier.verify(...)``` is following. ```indy_verifier_verify_proof``` only throws an error if there is something structurally/formatted incorrectly. Otherwise, it returns a bool.